src/map/utils/petutils.cpp

```diff
@@ -818,6 +826,8 @@ namespace petutils
             ref<uint16>(&PTrust->stats, counter) = (uint16)((raceStat + jobStat + sJobStat));
             counter += 2;
         }
+        PTrust->m_Weapons[SLOT_MAIN]->setDamage(GetTrustWeaponDamage(PTrust));
+        PTrust->m_Weapons[SLOT_MAIN]->setDelay((uint16)(floor(1000* (petStats->cmbDelay / 60))));		
```

Comment:
Is there a weapon type somewhere?

Reply:
These should just be getting their damage and delay from their mob_pools fields